lib/utils.c

```diff
@@ -71,15 +71,15 @@ int read_prop(const char *dev, char *prop, long *value)
 		*nl = '\0';
 
 	fclose(fp);
-	result = strtoul(buf, &endp, 0);
+	result = strtol(buf, &endp, 0);
 
 	if (*endp || buf == endp) {
 		fprintf(stderr, "value \"%s\" in file %s is not a number\n",
 			buf, fname);
 		goto out;
 	}
 
-	if (result == ULONG_MAX && errno == ERANGE) {
+	if (result == LONG_MAX && errno == ERANGE) {
```

Comment:
Since we're now converting a string to a signed quantity (`strtol()`), there can be overflow in either direction. So we also need to check whether result is `LONG_MIN`:

    if ((result == LONG_MAX || result == LONG_MIN) && errno == ERANGE) {